crates/ruma-common/src/api/error.rs

```diff
@@ -277,6 +277,10 @@ pub enum HeaderDeserializationError {
     #[error("missing header `{0}`")]
     MissingHeader(String),
 
+    /// The given header failed to parse.
+    #[error("invalid header: {0}")]
+    InvalidHeader(Box<dyn std::error::Error + Send + Sync + 'static>),
```

Comment:
I am not sure about this. We need a way to handle the errors returned from the `FromStr` implementations, but the `Send + Sync` bound might cause problems under WASM?

Reply:
I re-read RFC 6266 and found this paragraph:

> Recipients MAY take steps to recover a usable field value from an
   invalid header field, but SHOULD NOT reject the message outright,
   unless this is explicitly desirable behavior (e.g., the
   implementation is a validator).  As such, the default handling of
   invalid fields is to ignore them.

So I'm going to change the implementation of the `header` attribute to ignore errors when the field is an `Option`.